From c36d967915970fc057c2c8bb1a745a2b093eafb9 Mon Sep 17 00:00:00 2001
From: Jason Rogena <j.rogena@cgiar.org>
Date: Tue, 1 Apr 2014 10:44:50 +0300
Subject: [PATCH 137/213] Version 0.8 Revision 12: Users are now warned if they
 choose to go back to the main menu from an acivity where they entere data
 (AddEventActivity and AddMilkProductionActivity)

---
 MistroFarmer/src/main/AndroidManifest.xml            |  2 +-
 .../cgiar/ilri/mistro/farmer/AddEventActivity.java   | 19 ++++++++++++++++---
 .../mistro/farmer/AddMilkProductionActivity.java     | 20 +++++++++++++++++---
 .../cgiar/ilri/mistro/farmer/LandingActivity.java    |  2 +-
 .../java/org/cgiar/ilri/mistro/farmer/Utils.java     | 11 +++++++++++
 MistroFarmer/src/main/res/values/strings_en.xml      |  2 ++
 MistroFarmer/src/main/res/values/strings_sw.xml      |  2 ++
 7 files changed, 50 insertions(+), 8 deletions(-)

diff --git a/MistroFarmer/src/main/AndroidManifest.xml b/MistroFarmer/src/main/AndroidManifest.xml
index 1afcecc..da163f2 100644
--- a/MistroFarmer/src/main/AndroidManifest.xml
+++ b/MistroFarmer/src/main/AndroidManifest.xml
@@ -2,7 +2,7 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="org.cgiar.ilri.mistro.farmer"
     android:versionCode="2"
-    android:versionName="0.8.11" >
+    android:versionName="0.8.12" >
 
     <uses-sdk
         android:minSdkVersion="7"
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java
index a2ea083..a7d7ea2 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddEventActivity.java
@@ -209,9 +209,22 @@ public class AddEventActivity extends SherlockActivity implements View.OnClickLi
             return true;
         }
         else if(item.getItemId() == R.id.action_back_main_menu) {
-            Intent intent = new Intent(this, MainMenu.class);
-            intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
-            startActivity(intent);
+            DialogInterface.OnClickListener onClickListener = new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(DialogInterface dialog, int which) {
+                    if(which==DialogInterface.BUTTON_POSITIVE){
+                        dialog.dismiss();
+                        Intent intent = new Intent(AddEventActivity.this, MainMenu.class);
+                        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
+                        startActivity(intent);
+                    }
+                    else{
+                        dialog.cancel();
+                    }
+                }
+            };
+            AlertDialog mainMenuDialog = Utils.createMainMenuDialog(this, onClickListener);
+            mainMenuDialog.show();
         }
         return false;
     }
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java
index 0d9ccb1..c09d511 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java
@@ -1,5 +1,6 @@
 package org.cgiar.ilri.mistro.farmer;
 
+import android.app.AlertDialog;
 import android.app.DatePickerDialog;
 import android.app.Dialog;
 import android.app.ProgressDialog;
@@ -133,9 +134,22 @@ public class AddMilkProductionActivity extends SherlockActivity implements View.
             return true;
         }
         else if(item.getItemId() == R.id.action_back_main_menu) {
-            Intent intent = new Intent(this, MainMenu.class);
-            intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
-            startActivity(intent);
+            DialogInterface.OnClickListener onClickListener = new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(DialogInterface dialog, int which) {
+                    if(which==DialogInterface.BUTTON_POSITIVE){
+                        dialog.dismiss();
+                        Intent intent = new Intent(AddMilkProductionActivity.this, MainMenu.class);
+                        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
+                        startActivity(intent);
+                    }
+                    else{
+                        dialog.cancel();
+                    }
+                }
+            };
+            AlertDialog mainMenuDialog = Utils.createMainMenuDialog(this, onClickListener);
+            mainMenuDialog.show();
         }
         return false;
     }
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/LandingActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/LandingActivity.java
index 6940240..252b95a 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/LandingActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/LandingActivity.java
@@ -77,7 +77,7 @@ public class LandingActivity extends SherlockActivity implements View.OnClickLis
 
         //init text according to locale
         initTextInViews();
-        Toast.makeText(this, "Version 0.8 Revision 11", Toast.LENGTH_LONG).show();
+        Toast.makeText(this, "Version 0.8 Revision 12", Toast.LENGTH_LONG).show();
     }
 
     public boolean onCreateOptionsMenu(Menu menu) {
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/Utils.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/Utils.java
index bfd8f7a..e370dca 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/Utils.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/Utils.java
@@ -94,4 +94,15 @@ public class Utils
                 .setNegativeButton(Locale.getStringInLocale("cancel",context), onClickListener);
         return alertDialogBuilder.create();
     }
+
+    public static AlertDialog createMainMenuDialog(Context context, DialogInterface.OnClickListener onClickListener){
+        AlertDialog.Builder alertDialogBuilder=new AlertDialog.Builder(context);
+        alertDialogBuilder.setTitle(Locale.getStringInLocale("warning", context));
+        alertDialogBuilder
+                .setMessage(Locale.getStringInLocale("action_will_be_canceled", context))
+                .setCancelable(false)
+                .setPositiveButton(Locale.getStringInLocale("okay",context), onClickListener)
+                .setNegativeButton(Locale.getStringInLocale("cancel",context), onClickListener);
+        return alertDialogBuilder.create();
+    }
 }
diff --git a/MistroFarmer/src/main/res/values/strings_en.xml b/MistroFarmer/src/main/res/values/strings_en.xml
index ab32d40..802b732 100644
--- a/MistroFarmer/src/main/res/values/strings_en.xml
+++ b/MistroFarmer/src/main/res/values/strings_en.xml
@@ -229,4 +229,6 @@
     <string name="logout_en">Logout</string>
     <string name="back_en">Back</string>
     <string name="back_to_main_menu_en">Back to Main Menu</string>
+    <string name="warning_en">Warning</string>
+    <string name="action_will_be_canceled_en">If you choose to go back to the Main Menu the data you have entered in this screen will be lost. Press \'Okay\' if this is fine.</string>
 </resources>
diff --git a/MistroFarmer/src/main/res/values/strings_sw.xml b/MistroFarmer/src/main/res/values/strings_sw.xml
index a583493..27ee87c 100644
--- a/MistroFarmer/src/main/res/values/strings_sw.xml
+++ b/MistroFarmer/src/main/res/values/strings_sw.xml
@@ -229,4 +229,6 @@
     <string name="logout_sw">Toka</string>
     <string name="back_sw">Rudi</string>
     <string name="back_to_main_menu_sw">Rudi kwa Menu Kuu</string>
+    <string name="warning_sw">Onyo</string>
+    <string name="action_will_be_canceled_sw">Habari uliyoingiza katika ukurasa huu itapotea ukiamua kurudi kwa Menu Kuu. Bonyeza \'Sawa\' kama hii ni sawa na wewe.</string>
 </resources>
-- 
2.7.1

